tools/cert-sidecar/app/CronsCertsMgmt.js

```diff
@@ -0,0 +1,126 @@
+
+const { Logger, ConfigManager } = require('@dojot/microservice-sdk');
+const ServiceStateMgmt = require('./ServiceStateMgmt');
+const { cronJob } = require('./Utils');
+
+const {
+  cron: configCron,
+  certs: configCerts,
+  app: configApp,
+} = ConfigManager.getConfig('CERT_SC');
+
+const logger = new Logger(`cert-sc-${configApp['sidecar.to']}:CronsCertsMgmt`);
+
+
+/**
+* This class create some crons to getting news CRLs and check if
+* certificates are expiration and revoked.
+*/
+class CronsCertsMgmt {
+  /**
+   * Constructor CronsCertsMgmt
+   *
+   * @param {function} retrieveCRL
+   * @param {function} certsWillExpire
+   * @param {function} certHasRevoked
+   */
+  constructor(retrieveCRL, certsWillExpire, certHasRevoked) {
+    logger.debug('constructor: CronsMgmt');
+    this.retrieveCRLFunc = retrieveCRL;
+    this.certHasRevokedFunc = certHasRevoked;
+    this.certsWillExpireFunc = certsWillExpire;
+  }
+
+  /**
+   * Create crons to getting news CRLs and check if
+   * certificates are expiration and revoked.
+   */
+  initCrons() {
+    logger.info('initCrons: Initializing the crons...');
+
+    if (configCron.crl && configCerts.crl) {
+      this.cronUpdateCRL();
+    } else {
+      logger.info('initCrons: Cron for Update CRL is disabled');
+    }
+
+    if (configCron.expiration) {
+      this.cronCertsWillExpire();
+    } else {
+      logger.info('initCrons: Cron for Check Expiration is disabled');
+    }
+
+    if (configCron.revoke && configCerts.crl) {
+      this.cronCertHasRevoked();
+    } else {
+      logger.info('initCrons: Cron for Check has Revoke is disabled');
+    }
+
+    logger.info('initCrons: crons initialized');
+  }
+
+  /**
+   * Create a cron to check if certificate has revoke based in 'cron.revoke.time'
+   */
+  cronCertHasRevoked() {
+    logger.info('cronCertHasRevoked: Creating cron to check revoking...');
+    const boundCertHasRevoked = this.certHasRevokedFunc.bind(this);
+    cronJob(
+      async () => {
+        await boundCertHasRevoked()
+          .catch(async (e) => {
+            logger.error('Cron to certHasRevoked:', e);
+            // call shutdown in catch because the cron is running at other process
+            // and catch from index can't catch this exception
```

Comment:
Using CronJob by passing a callback function does not create a sub-process (via spawn object). Instead, the function is invoked in a setTimeout() in the event loop by the lib. I think this comment can be removed.